cirq/iqm/iqm_device.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2020 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""IQM devices https://iqm.fi/devices"""  # TODO: PQC-5
+
+from typing import cast
+import cirq
+from cirq import devices, ops
+
+
+class Adonis(devices.Device):
+    """IQM's five-qubit superconducting device with pairwise connectivity.
+    Details: https://iqm.fi/devices
+    """
+    # TODO: PQC-5
+
+    QUBIT_DIAGRAM = "-Q-\n" \
+                    "QQQ\n" \
+                    "-Q-\n"
+
+    SUPPORTED_GATES = (ops.CZPowGate, ops.ISwapPowGate, ops.XPowGate,
```

Comment:
Used only once for now. If it will not be used again, move to `validate_operation` method.

Reply:
Again, the idea is that this serves as class-public documentation about the supported gates. Would seem silly to just copy-paste this to a docstring.

Reply:
I think the point Victory is making is that this is not documentation, it is code.  The code is only used in validate_operation() so it does not need to be a class variable to have the same functionality.  If it is planned that this variable should be used elsewhere, it should be denoted as such.  If this is just for documentation, then it should be in documentation not in a variable, since the variable will not show up in pydoc, readthedocs, or in mouseovers for IDEs.

Reply:
Well, I still think that this tuple would be useful for developers to have as a public variable. But this is your project and I respect that, so I will encapsulate the list of supported gates inside the method that uses it and expose that information via a docstring.

Reply:
Done in https://github.com/quantumlib/Cirq/pull/2701/commits/d6eb9b535923873d7c880e82dd19802763e8b5e4